build_stage_repository

```diff
@@ -172,47 +168,167 @@ def packages_from_comps(comps):
     return packages
 
 
-def main():
-    try:
-        collection = sys.argv[1]
-        version = sys.argv[2]
-        dist = sys.argv[3]
-        arch = 'x86_64'
-    except IndexError:
-        raise SystemExit(f"Usage: {sys.argv[0]} collection version os")
+def copr_repository(collection, version, dist, arch):
+    return f"https://download.copr.fedorainfracloud.org/results/@theforeman/{collection}-{version}-staging/rhel-{dist.replace('el', '')}-{arch}"
 
-    base_dir = 'tmp'
-    rpm_sync_dir = f"{base_dir}/rpms"
-    srpm_sync_dir = f"{base_dir}/srpms"
 
-    stage_dir = f"{base_dir}/{collection}/{version}/{dist}/"
-    rpm_dir = f"{stage_dir}/{arch}"
-    srpm_dir = f"{stage_dir}/source"
+def prod_repository(collection, version, dist, arch):
+    if collection == "foreman":
+        subdir = "releases"
+    else:
+        subdir = collection.replace('foreman-', '')
 
-    if not os.path.exists(rpm_sync_dir):
-        os.makedirs(rpm_sync_dir)
+    return f"https://yum.theforeman.org/{subdir}/{version}/{dist}/{arch}"
 
-    if not os.path.exists(srpm_sync_dir):
-        os.makedirs(srpm_sync_dir)
 
-    if not os.path.exists(rpm_dir):
-        os.makedirs(rpm_dir)
+def copr_repository_urls(repository):
+    cmd = [
+        'dnf',
+        'reposync',
+        '--urls',
+        '--refresh',
+        '--repofrompath',
+        f"copr,{repository}",
+        '--repo',
+        'copr'
+    ]
 
-    if not os.path.exists(srpm_dir):
-        os.makedirs(srpm_dir)
+    urls = check_output(cmd, universal_newlines=True, stderr=STDOUT)
+    return urls.split("\n")
+
+
+def compare_repositories(new_repository, old_repository, source=False):
+    cmd = [
+        'dnf',
+        'repodiff',
+        '--simple',
+        '--refresh',
+        '--compare-arch',
+        '--repofrompath',
+        f"new,{new_repository}",
+        '--repofrompath',
+        f"old,{old_repository}",
+        '--repo-old',
+        'old',
+        '--repo-new',
+        'new'
+    ]
+
+    if not source:
+        cmd.extend(['--arch', 'noarch,x86_64'])
```

Comment:
In other places we have the arches as parameters. Just wondering, would it be
```suggestion
        cmd.extend(['--arch', f'noarch,{arch}'])
```

And then iterate that for every arch.